src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -96,5 +86,236 @@ public bool NeedsRefresh()
+        }
+
+        private FeatureFlag ParseFeatureFlag(string settingKey, string settingValue)
+        {
+            FeatureFlag featureFlag = new FeatureFlag();
+
+            var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(settingValue));
+
+            try
+            {
+                reader.Read();
+
+                if (reader.TokenType != JsonTokenType.StartObject)
+                {
+                    throw new FormatException(string.Format(ErrorMessages.FeatureFlagInvalidFormat, settingKey));
+                }
+
+                while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+                {
+                    if (reader.TokenType != JsonTokenType.PropertyName)
+                    {
+                        continue;
+                    }
+
+                    string propertyName = reader.GetString();
+
+                    switch (propertyName)
+                    {
+                        case FeatureManagementConstants.IdJsonPropertyName:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                                {
+                                    featureFlag.Id = reader.GetString();
+                                }
+                                else if (reader.TokenType == JsonTokenType.Null)
+                                {
+                                    // Effectively skips the feature flag if the Id is null
+                                    return null;
+                                }
+                                else
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.IdJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.String.ToString());
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.EnabledJsonPropertyName:
+                            {
+                                if (reader.Read() && (reader.TokenType == JsonTokenType.False || reader.TokenType == JsonTokenType.True))
+                                {
+                                    featureFlag.Enabled = reader.GetBoolean();
+                                }
+                                else
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.EnabledJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        $"{JsonTokenType.True}' or '{JsonTokenType.False}");
```

Comment:
The single quotes don't have matching pieces.

Reply:
With the error message it ends up looking correct because it has single quotes already around this string, but I can see how it looks odd here. Open to any suggestions for how to change this to be more clear.

Reply:
I see. Resolving.